ApollosProject/apollos-apps
# Up Next Feature  + Interactions Enhancements

## DESCRIPTION

### What does this PR do, or why is it needed?
Adds an algorithm that returns all series (essentially all content channel items with the children) where the user has completed at least one of the children. 

This PR also includes some code that will enable some functionality NewSpring was asking for. By extending the `Interactions` dataSource, you can provide additional methods that need to be called when a specific typename is interacted with via a specific action.  

![2020-05-01 12 25 00](https://user-images.githubusercontent.com/1637694/80821559-fc2f1f00-8ba6-11ea-946a-2a2da8683f7b.gif)

### What design trade-offs/decisions were made?

### How do I test this PR?

1. Add this feature to your config.yml `HOME_FEATURES` array. 
```
  - algorithms: [SERIES_IN_PROGRESS]
    subtitle: Continue
    type: HorizontalCardList
```
2. Visit the home feed. Note which series you see, and which series you don't see. 
3. Visit the discover tab and start another series by tapping the continue button or tapping on a child.
3. You should see that series show up in the home feed feature after a refresh. 


### What automated tests did you write?

working on that....

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
